plugins/uartdmx/UartDmxThread.cpp

```diff
@@ -72,11 +74,21 @@ bool UartDmxThread::WriteDMX(const DmxBuffer &buffer) {
  * The method called by the thread
  */
 void *UartDmxThread::Run() {
-  TimeStamp ts1, ts2;
-  Clock clock;
   CheckTimeGranularity();
   DmxBuffer buffer;
 
+  // basic frame time (without break): MAB + (time per bit * bits per slot * (slots per universe + 1)) + MALFT
+  // the +1 at the slots_per_universe is for the dmx start code
+  // the basic_frame_time is in microseconds
+  int basic_frame_time = DMX_MAB + (4 * 11 * (DMX_UNIVERSE_SIZE + 1)) + m_malft;
+
+  // the maximum time between breaks should be 1 second, and the minimum time 1204
+  // the capping of max and min values is done to honor the standard
+  int basic_frame_time_capped = std::max(std::min(basic_frame_time, 1000000), 1204);
+
+  // converts the frame time into milliseconds for later use
+  m_frameTime = static_cast<int>(floor(basic_frame_time_capped / static_cast<double>(1000)));
```

Comment:
I'd like to get rid of these magic numbers please and turn them into constants in the core of OLA as if we e.g. rewrote the FTDI plugin to match, it wants to use the same numbers, and things like the sniffer might want these values, indeed it already has their ranges:
https://github.com/OpenLightingProject/ola/blob/master/tools/logic/DMXSignalProcessor.cpp

There's some possible inspiration around naming here in another of our bits of software:
https://github.com/OpenLightingProject/ja-rule/blob/master/firmware/src/transceiver_timing.h

We've also got some generic clock related values too:
https://github.com/OpenLightingProject/ola/blob/master/include/ola/Clock.h

Reply:
That's a pretty good idea. I've replaced these magic numbers with constants.